### PR TITLE
[rc2] Throw when using multiple context types with non-generic DbContextOptions

### PR DESCRIPTION
**Pull request by natemcmaster**
Add #5017 to RC2. 
Fixes https://github.com/aspnet/EntityFramework/issues/4994

cc @ajcvickers @divega

### CODE REVIEW

**Comment by ajcvickers** [comment]
:shipit: (assuming ask mode approval)

File: src/Microsoft.EntityFrameworkCore/DbContextOptions`.cs [hunk 1]
```diff
@@ -56,5 +56,7 @@ public override DbContextOptions WithExtension<TExtension>(TExtension extension)
 
             return new DbContextOptions<TContext>(extensions);
         }
+
+        public override Type ContextType => typeof(TContext);
```
Comment (ajcvickers) [inline src/Microsoft.EntityFrameworkCore/DbContextOptions`.cs]: Presumably there will be another change to make use of this in Migrations?

Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore/DbContextOptions`.cs]: Yes, I'll merge this first and then rebase the other PR on top of this.

**Comment by plamenyovchev** [comment]
Hello guys, I am experiencing the same issue with with version of .NET Core 2.1 SDK  and EntityFramework Core 2.1.4

My DbContexts are: 


```
public class LarMasterDbContext : DbContext
    {
        public DbSet<ClientConnectionString> ClientConnectionStrings { get; set; }

        public LarMasterDbContext(DbContextOptions<LarMasterDbContext> options)
            :base(options)
        {
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            base.OnModelCreating(modelBuilder);
        }
    }
```


and 

```
 public class LarDbContext : IdentityDbContext<LarUser>
    {
        public LarDbContext(DbContextOptions<LarDbContext> options)
            :base(options)
        {
        }

        protected override void OnModelCreating(ModelBuilder builder)
        {
            base.OnModelCreating(builder);
        }
    }
```

After executing Add-Migration InitialCreate -Context LarMasterDbContext -OutputDir Migrations\MasterDb

I am getting Object reference not set to an instance of an object. in LarDbContext. It is configured on second place in my Startup.cs 


```
services.AddDbContext<LarMasterDbContext>(
                opt => opt.UseMySql(Configuration.GetConnectionString("MasterDbConnection")));

            services.AddHttpContextAccessor();
            services.AddDbContext<LarDbContext>((serviceProvider, opt) =>
            {
                var httpRequest = serviceProvider.GetService<IHttpContextAccessor>().HttpContext.Request;
                using (var masterDb = serviceProvider.GetService<LarMasterDbContext>())
                {
                    // TO DO: need to check for missing client code
                    var clientCode = httpRequest.Headers["clientCode"].ToString();
                    var connection = masterDb.ClientConnectionStrings.FirstOrDefault(x => x.ClientCode == clientCode)?.ConnectionString;
                    opt.UseMySql(connection);
                };
            });
```


Am I doing something wrong ? If I comment the second add of DbContext the command executes successfully and migration is created for me.

**Pull request merged**